runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/artifact/BeamFileSystemArtifactStagingService.java

```diff
@@ -77,6 +78,13 @@
   public static final String MANIFEST = "MANIFEST";
   public static final String ARTIFACTS = "artifacts";
 
+  private final Semaphore permittedConcurrentWrite;
+
+  public BeamFileSystemArtifactStagingService() {
+    super();
+    permittedConcurrentWrite = new Semaphore(8);
```

Comment:
Maybe make the `8` a constant?

Reply:
The name of the variable `permittedConcurrentWrite` seems to be descriptive enough hence i avoided creating a new constant.
Let me know if you think it will be better to create a constant.